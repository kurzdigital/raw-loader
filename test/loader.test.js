import crypto from 'crypto';

import {
  compile,
  execute,
  getCompiler,
  normalizeErrors,
  readAsset,
} from './helpers';

describe('loader', () => {
  it('should work', async () => {
    const compiler = getCompiler('simple.js');
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should not work with PDF without raw option', async () => {
    const compiler = getCompiler('pdf.js');
    const stats = await compile(compiler);

    expect(
      crypto
        .createHash('md5')
        .update(execute(readAsset('main.bundle.js', compiler, stats)))
        .digest('hex')
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with PDF with raw option', async () => {
    const compiler = getCompiler('pdf.js', { raw: true });
    const stats = await compile(compiler);

    expect(
      crypto
        .createHash('md5')
        .update(
          Buffer.from(execute(readAsset('main.bundle.js', compiler, stats)))
        )
        .digest('hex')
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should not work with cp1251 without raw option', async () => {
    const compiler = getCompiler('cp1251.js');
    const stats = await compile(compiler);

    expect(
      Array.from(
        execute(readAsset('main.bundle.js', compiler, stats))
      ).map((c) => c.charCodeAt(0))
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with cp1251 with raw option', async () => {
    const compiler = getCompiler('cp1251.js', { raw: true });
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should not work with binary', async () => {
    const compiler = getCompiler('binary.js');
    const stats = await compile(compiler);

    expect(
      crypto
        .createHash('md5')
        .update(execute(readAsset('main.bundle.js', compiler, stats)))
        .digest('hex')
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with binary with raw option', async () => {
    const compiler = getCompiler('binary.js', { raw: true });
    const stats = await compile(compiler);

    expect(
      crypto
        .createHash('md5')
        .update(
          Buffer.from(execute(readAsset('main.bundle.js', compiler, stats)))
        )
        .digest('hex')
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with inline loader', async () => {
    const compiler = getCompiler('inline.js');
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with "ModuleConcatenationPlugin" plugin', async () => {
    const compiler = getCompiler(
      'simple.js',
      {},
      {
        mode: 'production',
        optimization: {
          minimize: false,
        },
      }
    );
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');

    if (stats.compilation.modules.size) {
      expect(stats.compilation.modules.size).toBe(1);
    } else {
      expect(stats.compilation.modules.length).toBe(1);
    }
  });
});
