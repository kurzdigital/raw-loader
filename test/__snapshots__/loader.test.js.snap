// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader should not work with PDF without raw option: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should not work with PDF without raw option: errors 1`] = `Array []`;

exports[`loader should not work with PDF without raw option: result 1`] = `"c72a71c5990f1caedaa9be40a547be9f"`;

exports[`loader should not work with PDF without raw option: warnings 1`] = `Array []`;

exports[`loader should not work with binary: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should not work with binary: errors 1`] = `Array []`;

exports[`loader should not work with binary: result 1`] = `"fbfa5723382326d54d66958764496c83"`;

exports[`loader should not work with binary: warnings 1`] = `Array []`;

exports[`loader should not work with cp1251 without raw option: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should not work with cp1251 without raw option: errors 1`] = `Array []`;

exports[`loader should not work with cp1251 without raw option: result 1`] = `
Array [
  65533,
  65533,
  65533,
  32,
  65533,
  65533,
  65533,
  65533,
  32,
  65533,
  32,
  65533,
  65533,
  65533,
  32,
  65533,
  32,
  65533,
  65533,
  65533,
  65533,
  65533,
  65533,
  32,
  65533,
  10,
  65533,
  65533,
  65533,
  65533,
  65533,
  32,
  65533,
  65533,
  65533,
  65533,
  65533,
  32,
  65533,
  65533,
  65533,
  65533,
  65533,
  65533,
  46,
  10,
  10,
  65533,
  65533,
  65533,
  65533,
  65533,
  65533,
  65533,
  32,
  65533,
  65533,
  65533,
  65533,
  65533,
  65533,
  65533,
  65533,
  65533,
  65533,
  32,
  65533,
  65533,
  65533,
  65533,
  65533,
  65533,
  65533,
  65533,
  10,
  10,
  87,
  104,
  101,
  114,
  101,
  32,
  116,
  104,
  101,
  32,
  114,
  111,
  115,
  101,
  115,
  32,
  97,
  114,
  101,
  32,
  45,
  32,
  116,
  104,
  101,
  114,
  101,
  32,
  97,
  114,
  101,
  32,
  116,
  104,
  111,
  114,
  110,
  115,
  32,
  45,
  10,
  84,
  104,
  97,
  116,
  32,
  105,
  115,
  32,
  116,
  104,
  101,
  32,
  108,
  97,
  119,
  32,
  111,
  102,
  32,
  102,
  97,
  116,
  101,
  46,
  10,
  10,
  78,
  105,
  107,
  111,
  108,
  97,
  121,
  32,
  65,
  108,
  101,
  107,
  115,
  101,
  101,
  118,
  105,
  99,
  104,
  32,
  78,
  101,
  107,
  114,
  97,
  115,
  111,
  118,
  10,
]
`;

exports[`loader should not work with cp1251 without raw option: warnings 1`] = `Array []`;

exports[`loader should work with "ModuleConcatenationPlugin" plugin: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should work with "ModuleConcatenationPlugin" plugin: errors 1`] = `Array []`;

exports[`loader should work with "ModuleConcatenationPlugin" plugin: result 1`] = `
"Где розы — там и тернии —
Таков закон судьбы.

Николай Алексеевич Некрасов

Where the roses are - there are thorns -
That is the law of fate.

Nikolay Alekseevich Nekrasov
"
`;

exports[`loader should work with "ModuleConcatenationPlugin" plugin: warnings 1`] = `Array []`;

exports[`loader should work with PDF with raw option: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should work with PDF with raw option: errors 1`] = `Array []`;

exports[`loader should work with PDF with raw option: result 1`] = `"f9c0d106c7d10f1b60a857bd08438f93"`;

exports[`loader should work with PDF with raw option: warnings 1`] = `Array []`;

exports[`loader should work with binary with raw option: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should work with binary with raw option: errors 1`] = `Array []`;

exports[`loader should work with binary with raw option: result 1`] = `"394e219db24b8dfb02b7db40543c81f2"`;

exports[`loader should work with binary with raw option: warnings 1`] = `Array []`;

exports[`loader should work with cp1251 with raw option: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should work with cp1251 with raw option: errors 1`] = `Array []`;

exports[`loader should work with cp1251 with raw option: result 1`] = `
Object {
  "data": Array [
    195,
    228,
    229,
    32,
    240,
    238,
    231,
    251,
    32,
    151,
    32,
    242,
    224,
    236,
    32,
    232,
    32,
    242,
    229,
    240,
    237,
    232,
    232,
    32,
    151,
    10,
    210,
    224,
    234,
    238,
    226,
    32,
    231,
    224,
    234,
    238,
    237,
    32,
    241,
    243,
    228,
    252,
    225,
    251,
    46,
    10,
    10,
    205,
    232,
    234,
    238,
    235,
    224,
    233,
    32,
    192,
    235,
    229,
    234,
    241,
    229,
    229,
    226,
    232,
    247,
    32,
    205,
    229,
    234,
    240,
    224,
    241,
    238,
    226,
    10,
    10,
    87,
    104,
    101,
    114,
    101,
    32,
    116,
    104,
    101,
    32,
    114,
    111,
    115,
    101,
    115,
    32,
    97,
    114,
    101,
    32,
    45,
    32,
    116,
    104,
    101,
    114,
    101,
    32,
    97,
    114,
    101,
    32,
    116,
    104,
    111,
    114,
    110,
    115,
    32,
    45,
    10,
    84,
    104,
    97,
    116,
    32,
    105,
    115,
    32,
    116,
    104,
    101,
    32,
    108,
    97,
    119,
    32,
    111,
    102,
    32,
    102,
    97,
    116,
    101,
    46,
    10,
    10,
    78,
    105,
    107,
    111,
    108,
    97,
    121,
    32,
    65,
    108,
    101,
    107,
    115,
    101,
    101,
    118,
    105,
    99,
    104,
    32,
    78,
    101,
    107,
    114,
    97,
    115,
    111,
    118,
    10,
  ],
  "type": "Buffer",
}
`;

exports[`loader should work with cp1251 with raw option: warnings 1`] = `Array []`;

exports[`loader should work with inline loader: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should work with inline loader: errors 1`] = `Array []`;

exports[`loader should work with inline loader: result 1`] = `
"Где розы — там и тернии —
Таков закон судьбы.

Николай Алексеевич Некрасов

Where the roses are - there are thorns -
That is the law of fate.

Nikolay Alekseevich Nekrasov
"
`;

exports[`loader should work with inline loader: warnings 1`] = `Array []`;

exports[`loader should work: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should work: errors 1`] = `Array []`;

exports[`loader should work: result 1`] = `
"Где розы — там и тернии —
Таков закон судьбы.

Николай Алексеевич Некрасов

Where the roses are - there are thorns -
That is the law of fate.

Nikolay Alekseevich Nekrasov
"
`;

exports[`loader should work: warnings 1`] = `Array []`;
